modules/web/middleware/locale.go

```diff
@@ -46,7 +47,15 @@ func Locale(resp http.ResponseWriter, req *http.Request) translation.Locale {
 		SetLocaleCookie(resp, lang, 1<<31-1)
 	}
 
-	return translation.NewLocale(lang)
+	err := translation.NewLocale(lang)
+	if err != nil {
+		return err
+	}
+
+	if skipHandler {
```

Comment:
I haven't quite understood what the benefit of this skip handler is?
Why would you name some weird boolean that does nothing else than send a `204` `skipHandler`?
Why does that fix the bug?
Why is that return value even required for AJAX?
Also, if you only use it here, why not inline the variable?
Please document all that in the backend code as well.

Reply:
Alternatively, as you wrote yourself, we could also fix the incorrect behavior with the new one you proposed.
I think I'm in favor of your proposal as it seems way cleaner and more understandable.

Reply:
>  Why would you name some weird boolean that does nothing else than send a 204 skipHandler?
>  Why does that fix the bug?

the irrelevant POST requests were processed by current page's handler. 

When switching the language, the handler will also receive an empty form, it will lead to some incorrect results in some cases (like the installation page, it will reset some default values)

Reply:
After some more  research, I think the old behavior should still be kept.

The full logic is:
* Current page `/foo/bar?a=b`
* Language menu: `/?lang=xx`
* If a crawler comes, it follows `/?lang=xx`
* If  a user clicks the menu, it first requests `/?lang=xx&lang_skip_handler=1` and then reload current page `/foo/bar?a=b`